FSW/src/headers/encodedSciData.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef ENCODEDSCI_H
+#define ENCODEDSCI_H 
+
+/* - - - - - - Includes - - - - - - */
+// C++ libraries
+
+// Other libraries
+
+// NS2 config and utility headers
+#include "config.hpp"
+#include "encodedFile.hpp"
+
+
+/* - - - - - - Class Declaration - - - - - - */
+
+class EncodedSciData : public EncodedFile<SCIDATA_RAW_MEMSIZE> {
+    private:
+        // member variables
+        uint16_t m_buffer[BUFFERSIZE];        // array to hold decoded buffer
+        unsigned long m_timestamp = MEMSIZE;  // file timestamp
+
+    public:
+        // constructors
+        EncodedSciData() { }
+        EncodedSciData(uint16_t *buffer, unsigned long &timestamp);
```

Comment:
The default constructor is used when we want to fill the `encodedFile` with data that is already encoded.  An example is memory scrubbing; we construct and empty `encodedFile` with the default constructor and then call `fill(src)` which fills the object with data from `src` so it can be scrubbed.

Reply:
gotcha, thanks for the explanation